ACE/ace/CDR_Stream.cpp

```diff
@@ -1707,14 +1707,18 @@ ACE_InputCDR::read_string (std::string& x)
   // the memory is allocated.
   if (len > 0 && len <= this->length())
     {
+#if !defined (ACE_NO_GHS_EXCEPT)
```

Comment:
Assuming this macro is new, maybe it could have a better name? Like `ACE_LACKS_EXCEPTIONS` or `ACE_LACKS_STD_EXCEPTIONS`? I don't know what the platform supports, so I don't know what I'm the scope of this is.

Reply:
How does integrity handle a resize error? Shouldn't be there some alternative code instead to handle the error cases?

Reply:
ACE/TAO master requires C++ exception support, support for emulated exceptions was removed years ago

Reply:
One of the design principles of ACE is avoiding exception handling: https://github.com/DOCGroup/ACE_TAO/blob/master/ACE/docs/ACE-porting.html#L158-L163. Requiring C++ exceptions in ACE seems to go against it. In some real-time systems which use ACE, it's desirable to turn off exception support to improve determinism.

Reply:
That documentation is outdated as far as I know, C++ exception support is I think mandatory for years now, will dig tomorrow briefly in the ACE history to see from when

Reply:
Support for non exception builds was removed in 2010, looks the docu need an update https://github.com/DOCGroup/ACE_TAO/blob/23b4b06cc754916a678f207e234502ffbe35ee2d/ACE/ChangeLogs/ChangeLog-2011a#L5371

Reply:
When will this code even be used?  TAO doesn't use `std::string` in the IDL mapping.
Assuming it's not needed, we can just exclude it from the embedded system builds.

Reply:
It is used by the alternate STL mapping support in TAO_IDL as far as I know and it is used by TAOX11/AXCIOMA in various places

Reply:
We will exclude it on this platform/configuration, which doesn't support that std library function.

Reply:
Actually I got that wrong -- the std library function is supported but error handling is different (std::allocator behaves differently).  @sonndinh will update the code.